### PR TITLE
Add Bloodhound service for GeoAdmin location search

### PR DESCRIPTION
**Pull request by tsauerwein**
This PR proposes to add a service for the [GeoAdmin location search](http://api3.geo.admin.ch/services/sdiservices.html#search) which creates a Bloodhound object that parses the response from the location service and creates features (similar to the `createGeoJSONBloodhound` service).

### CODE REVIEW

File: examples/locationsearch.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+/**
+ * @param {jQuery.Event} event Event.
+ * @param {Object} suggestion Suggestion.
+ * @param {TypeaheadDataset} dataset Dataset.
+ * @this {app.SearchController}
+ * @private
+ */
+app.SearchController.select_ = function(event, suggestion, dataset) {
+  var map = /** @type {ol.Map} */ (this.map);
```
Comment (gberaudo) [inline examples/locationsearch.js]: Is it necessary to cast the map?

File: examples/locationsearch.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  var feature = /** @type {ol.Feature} */ (suggestion);
+  var bbox = /** @type {ol.Extent} */
+      (feature.get('bbox'));
+  var mapSize = /** @type {ol.Size} */ (map.getSize());
```
Comment (gberaudo) [inline examples/locationsearch.js]: Have you considered using an assert instead of casting?

File: examples/locationsearch.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+  map.getView().fit(bbox, mapSize,
+      /** @type {olx.view.FitOptions} */ ({maxZoom: 16}));
```
Comment (gberaudo) [inline examples/locationsearch.js]: Is this cast necessary?

File: src/modules/search/createlocationsearchbloodhound.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        parseFloat(match[3]),
+        parseFloat(match[4])
+      ];
+    } else {
+      return null;
+    }
+  };
+
+  var removeHtmlTags = function(label) {
+    return label.replace(/<\/?[ib]>/g, '');
```
Comment (gberaudo) [inline src/modules/search/createlocationsearchbloodhound.js]: Shouldn't you remove all tags and not only b and i?

Comment (tsauerwein) [inline src/modules/search/createlocationsearchbloodhound.js]: `<b>` and `<i>` are the only tags used.

File: examples/locationsearch.js (hunk 1 above)
Comment (tsauerwein) [inline examples/locationsearch.js]: Thanks for noticing, there were quite some unneeded casts in this example.

**Comment by tsauerwein** [comment]
Thanks for the review!

**Pull request merged**